lib/livebook_web/live/output/input_component.ex

```diff
@@ -305,6 +367,35 @@ defmodule LivebookWeb.Output.InputComponent do
     {:ok, html_value}
   end
 
+  defp parse(html_value, %{type: :datetime}) do
+    case NaiveDateTime.from_iso8601(html_value) do
+      {:ok, datetime} -> {:ok, datetime}
+      {:error, _error} -> :error
+    end
+  end
+
+  defp parse(html_value, %{type: :time}) do
+    case Time.from_iso8601(html_value) do
+      {:ok, time} -> {:ok, time}
+      {:error, _error} -> :error
+    end
+  end
+
+  defp parse(html_value, %{type: :date}) do
+    case Date.from_iso8601(html_value) do
```

Comment:
We can't specify milliseconds in the input, so I think we should make sure we parse into datetime with second precision.